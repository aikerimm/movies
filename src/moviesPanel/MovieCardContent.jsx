import PropTypes from 'prop-types';
import React from 'react';

const MovieCardContent = ({ movie, onMovieClick }) => {
  return (
    <>
      <img
        src={movie.imageName}
        alt='moviePoster'
        className='movieCardPoster'
        onClick={() => onMovieClick(movie)}
      />
      <div className='titleYear'>
        <p className='movieCardTitle'>{movie.title}</p>
        <p className='movieCardYear'>{movie.releaseDate.getFullYear()}</p>
      </div>
      <p className='movieCardGenre'>{movie.genre}</p>
    </>
  );
};

MovieCardContent.propTypes = {
  movie: PropTypes.shape({
    title: PropTypes.string.isRequired,
    imageName: PropTypes.string.isRequired,
    releaseDate: PropTypes.instanceOf(Date).isRequired,
    genre: PropTypes.arrayOf(PropTypes.string).isRequired,
  }),
  onMovieClick: PropTypes.func.isRequired,
};

export default React.memo(MovieCardContent);
