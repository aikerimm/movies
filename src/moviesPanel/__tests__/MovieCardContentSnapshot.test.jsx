import renderer from 'react-test-renderer';
import MovieCardContent from '../MovieCardContent';

it('MovieCardContent should render correctly', () => {
  const movie = {
    id: 123,
    release_date: '1995-01-01',
    title: 'Dune',
    genres: ['Action', 'Drama'],
    poster_path: '',
  };
  const tree = renderer
    .create(<MovieCardContent movie={movie} onMovieCardClick={jest.fn()} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
