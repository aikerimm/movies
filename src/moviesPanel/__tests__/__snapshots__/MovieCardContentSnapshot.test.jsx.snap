// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieCardContent should render correctly 1`] = `
[
  <img
    alt="moviePoster"
    className="movieCardPoster"
    onClick={[MockFunction]}
    src=""
  />,
  <div
    className="titleYear"
  >
    <p
      className="movieCardTitle"
    >
      Dune
    </p>
    <p
      className="movieCardYear"
    >
      1995
    </p>
  </div>,
  <p
    className="movieCardGenre"
  >
    Action, Drama
  </p>,
]
`;
